### PR TITLE
update dashboard to support communicating with supervisord

### PR DESCRIPTION
**Pull request by snopoke**
@TylerSheffels

### CODE REVIEW

**Comment by snopoke** [comment]
I still want to go over this once more and also add some doc strings etc.

File: corehq/util/supervisord/api.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+    def get_status(pillow_name):
+        try:
+            info = supervisor.get_pillow_process_info(pillow_name)
+            return status(info.statename)
+        except Exception as e:
+            return status('UNKNOWN', str(e))
+
+    return {name: get_status(name) for name in pillow_names}
+
```
Comment (dimagimon) [inline corehq/util/supervisord/api.py]: W391 blank line at end of file

**Comment by dannyroberts** [comment]
Super cool Simon!

**Comment by czue** [comment]
nice!

File: corehq/util/supervisord/api.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
+import os
```
Comment (dimagimon) [inline corehq/util/supervisord/api.py]: F401 'os' imported but unused

File: corehq/util/supervisord/api.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+import xmlrpclib
+from amqplib.client_0_8.method_framing import defaultdict
+from ansible.inventory import InventoryParser
```
Comment (dimagimon) [inline corehq/util/supervisord/api.py]: F401 'InventoryParser' imported but unused

**Comment by snopoke** [comment]
:white_check_mark: GTG

**Comment by dannyroberts** [comment]
:tada: :balloon:

**Comment by czue** [comment]
super cool.

**Pull request merged**